### ISSUE TITLE
hara.io.scheduler - schedule task for the last day of the month

### ISSUE THREAD

**Issue by foomip** [zcaudate/hara#39]
Not sure if anyone has given some thought to this feature. It would be great if there was a way to represent the last day of the month (which could be 28,29,30 or 31) in the cron string for the scheduler.

For example I want a task to execute on the last day of the month at 3:05am - so represent that with the cron string `* 3 5 * L * *` - so the character L representing the last day of the month.
 
Is someone currently working on a feature like this, or is something like this planned for implementation?

**Comment by zcaudate** [comment]
I like the idea... it's not too complicated to add in as the controls for validating the cron string are here:
https://github.com/zcaudate/hara/blob/master/src/hara/io/scheduler/tab.clj

I'm just not sure of this feature because of various things it needs to support like leap years. It's supposed to be pretty lightweight so having calendar properties messes things up.

**Comment by zcaudate** [comment]
hmm... it is there for quartz: http://www.quartz-scheduler.org/documentation/quartz-2.1.x/tutorials/crontrigger

**Comment by foomip** [comment]
Yes I see they also use "L" for last day of the week and "W" for every weekday.

**Comment by foomip** [comment]
I think you could keep it as simple modifying the `match-array?` function to parse out the L and W string with the current month's/week's relevant integer value just before initiating the comparison?

**Comment by zcaudate** [comment]
Right now, `match-array` is not context aware - which it needs to be if additional features such as the ones on quartz needs to be supported. I remember a similar discussion here: https://github.com/zcaudate/hara/issues/17

Unfortunately I just don't have bandwidth to implement this right now.

**Comment by zcaudate** [comment]
If someone wants to have a go at the implementation, the time and date checkers need to be somehow broken up in order to maintain good efficiency but still be practical. It's something that needs a bit more thought.
